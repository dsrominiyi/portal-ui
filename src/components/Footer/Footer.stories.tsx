import React from 'react';
import { storiesOf } from '@storybook/react';

import Footer from '.';

const stories = storiesOf('Footer', module);

const render = (copyrightYear: number) => () => <Footer copyrightYear={copyrightYear} />;

export const storyRenders = {
  default: render(2019)
};

stories.add('Copyright Year supplied', storyRenders.default);
