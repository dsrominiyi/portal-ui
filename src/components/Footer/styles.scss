@import '../../style/vars';

.footer {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 70px;
  width: 100%;
  background-color: $brand-white;
  font-size: 0.8rem;
}
