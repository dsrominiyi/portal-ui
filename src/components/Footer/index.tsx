import React from 'react';

import styles from './styles.scss';

interface Props {
  copyrightYear: number;
}

export const Footer: React.FunctionComponent<Props> = ({ copyrightYear }) => (
  <div className={styles.footer}>{`Copyright \u00A9 Datamatters ${copyrightYear}`}</div>
);

Footer.displayName = 'Footer';
export default Footer;
