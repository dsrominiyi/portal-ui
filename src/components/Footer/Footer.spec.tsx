import { storyRenders } from './Footer.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('Footer', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
