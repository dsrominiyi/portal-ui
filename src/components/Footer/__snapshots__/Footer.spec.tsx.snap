// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component story renders match snapshots: default 1`] = `
<div
  className="footer"
>
  Copyright Â© Datamatters 2019
</div>
`;
